### PR TITLE
Documentation for variadics

### LINKED ISSUE #15208: Add fake variadics to `Reflect` impls
## How can Bevy's documentation be improved?

@bash added lovely fake variadics for the documentation of `all_tuples!`. See https://github.com/bevyengine/bevy/issues/14697

The same can be done for `impl_reflect_tuple!` :) 

Anyone tackling this can just copy the approach taken in the linked issue, so this *should* be straightforward

### PR DESCRIPTION
**Pull request by BenjaminBrienen**
# Objective

Relevant: #15208

## Solution

I went ahead and added the variadics documentation in all applicable locations.

## Testing

- I built the documentation and inspected it to see whether the feature is there.

### CODE REVIEW

**Comment by BenjaminBrienen** [comment]
@bash do you know why this isn't working? I added the marker attribute but the documentation doesn't have the new look.

**Comment by bash** [comment]
@BenjaminBrienen What command did you use to build the docs? There's some `RUSTFLAGS` / `RUSTDOCFLAGS` needed. See https://github.com/bevyengine/bevy/pull/14820#issuecomment-2313440715

**Comment by bash** [comment]
Btw `#[doc(fake_variadic)]` only works on "variadic" tuples or "variadic" `fn` pointers. It doesn't work more complex types like `SystemState<(T1, T2, ...)` unfortunately.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by MrGVSV** [comment]
> Btw `#[doc(fake_variadic)]` only works on "variadic" tuples or "variadic" `fn` pointers. It doesn't work more complex types like `SystemState<(T1, T2, ...)` unfortunately.

Yeah I'm getting an error when attempting to build locally:

```
error: `#[doc(fake_variadic)]` must be used on the first of a set of tuple or fn pointer trait impls with varying arity
   --> crates/bevy_reflect/src/func/info.rs:334:1
    |
334 | / all_tuples!(
335 | |     #[doc(fake_variadic)]
336 | |     impl_typed_function,
337 | |     0,
...   |
340 | |     arg
341 | | );
    | |_^
    |
    = note: this error originates in the macro `all_tuples` (in Nightly builds, run with -Z macro-backtrace for more info)
```

File: crates/bevy_reflect/src/tuple.rs
Comment (MrGVSV) [inline crates/bevy_reflect/src/tuple.rs:1]: We should probably update the `impl_reflect_tuple!` macro in this file to make use of this attribute as well.

**Review by MrGVSV — CHANGES_REQUESTED** [review]
CI is passing, but I don't know if it builds with the right flags set. Some of these are failing to build docs (likely due to the type being annotated like @bash mentioned).

File: crates/bevy_reflect/src/tuple.rs
Comment (BenjaminBrienen) [reply crates/bevy_reflect/src/tuple.rs:1]: I am not good enough with macros to fix the ambiguity in the macro match. Can you help me with that?

Comment (BenjaminBrienen) [reply crates/bevy_reflect/src/tuple.rs:1]: > We should probably update the `impl_reflect_tuple!` macro in this file to make use of this attribute as well.

It's not straightforward for me, so I'm just going to let that be a seperate PR if that's ok.

File: crates/bevy_ecs/src/system/exclusive_function_system.rs [hunk 1]
```diff
@@ -216,7 +216,7 @@ pub trait ExclusiveSystemParamFunction<Marker>: Send + Sync + 'static {
 pub struct HasExclusiveSystemInput;
 
 macro_rules! impl_exclusive_system_function {
-    ($($param: ident),*) => {
+    ($(#[$meta:meta])* $($param: ident),*) => {
```
Comment (pablo-lua) [inline crates/bevy_ecs/src/system/exclusive_function_system.rs:219]: I noticed that even though this was added, there is no use of this new var, is it right?

File: crates/bevy_reflect/src/tuple.rs
Comment (MrGVSV) [reply crates/bevy_reflect/src/tuple.rs:1]: Yeah I think this requires a larger change to `all_tuples`. I’m playing around with it now

File: crates/bevy_reflect/src/func/reflect_fn_mut.rs [hunk 2]
```diff
@@ -201,4 +202,11 @@ macro_rules! impl_reflect_fn_mut {
     };
 }
 
-all_tuples!(impl_reflect_fn_mut, 0, 15, Arg, arg);
+all_tuples!(
```
Comment (MrGVSV) [inline crates/bevy_reflect/src/func/reflect_fn_mut.rs:205]: Testing locally, these ones still seem to throw an error (unless I’m using the wrong command)

**Review by MrGVSV — CHANGES_REQUESTED** [review]

**Comment by BenjaminBrienen** [comment]
@alice-i-cecile I'm ready for this to be reviewed again

> > Btw `#[doc(fake_variadic)]` only works on "variadic" tuples or "variadic" `fn` pointers. It doesn't work more complex types like `SystemState<(T1, T2, ...)` unfortunately.
> 
> Yeah I'm getting an error when attempting to build locally:
> 
> ```
> error: `#[doc(fake_variadic)]` must be used on the first of a set of tuple or fn pointer trait impls with varying arity
>    --> crates/bevy_reflect/src/func/info.rs:334:1
>     |
> 334 | / all_tuples!(
> 335 | |     #[doc(fake_variadic)]
> 336 | |     impl_typed_function,
> 337 | |     0,
> ...   |
> 340 | |     arg
> 341 | | );
>     | |_^
>     |
>     = note: this error originates in the macro `all_tuples` (in Nightly builds, run with -Z macro-backtrace for more info)
> ```
```
PS C:\Users\BenjaminBrienen\source\bevy> set RUSTFLAGS='--cfg docsrs_dep' && set RUSTDOCFLAGS='--cfg=docsrs' && cargo +nightly doc
    Updating crates.io index
     Locking 6 packages to latest compatible versions
      Adding assert_type_match v0.1.1
    Updating encase v0.9.0 -> v0.10.0
    Updating encase_derive v0.9.0 -> v0.10.0
      Adding encase_derive_impl v0.10.0
    Updating glam v0.28.0 -> v0.29.0
    Updating hexasphere v14.1.0 -> v15.0.0
    Blocking waiting for file lock on package cache
    Checking bevy_ptr v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ptr)
    Checking disqualified v1.0.0
   Compiling bevy_macro_utils v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_macro_utils)
   Compiling bevy_utils_proc_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_utils\macros)
   Compiling assert_type_match v0.1.1
    Checking bevy_tasks v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_tasks)
    Checking glam v0.29.0
 Documenting bevy_macro_utils v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_macro_utils)
 Documenting glam v0.29.0
 Documenting bevy_utils_proc_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_utils\macros)
 Documenting assert_type_match v0.1.1
 Documenting disqualified v1.0.0
 Documenting bevy_ptr v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ptr)
   Compiling encase_derive_impl v0.10.0
 Documenting bevy_tasks v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_tasks)
   Compiling rustix v0.38.37
    Checking errno v0.3.9
 Documenting encase_derive_impl v0.10.0
    Checking polling v3.7.3
    Checking bytes v1.7.1
    Checking pin-utils v0.1.0
    Checking futures-task v0.3.30
   Compiling httparse v1.9.4
    Checking bevy_utils v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_utils)
   Compiling bevy_reflect_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_reflect\derive)
   Compiling bevy_ecs_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ecs\macros)
   Compiling bevy_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_derive)
   Compiling encase_derive v0.10.0
   Compiling bevy_asset_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_asset\macros)
   Compiling bevy_encase_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_encase_derive)
   Compiling bevy_render_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_render\macros)
    Checking http v1.1.0
    Checking futures-util v0.3.30
 Documenting errno v0.3.9
    Checking futures-channel v0.3.30
    Checking tokio v1.40.0
    Checking httpdate v1.0.3                                                                                                                                                                                       
 Documenting bevy_reflect_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_reflect\derive)
 Documenting bevy_ecs_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ecs\macros)                                                                                                             
 Documenting bevy_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_derive)                                                                                                                     
 Documenting bevy_utils v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_utils)                                                                                                                       
 Documenting bevy_asset_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_asset\macros)                                                                                                         
 Documenting bevy_render_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_render\macros)
    Checking async-io v2.3.4                                                                                                                                                                                       
    Checking http-body v1.0.1                                                                                                                                                                                      
 Documenting polling v3.7.3
   Compiling anyhow v1.0.89                                                                                                                                                                                        
 Documenting bytes v1.7.1                                                                                                                                                                                          
    Checking hyper v1.4.1
    Checking async-process v2.3.0                                                                                                                                                                                  
    Checking async-net v2.0.0
 Documenting futures-task v0.3.30                                                                                                                                                                                  
 Documenting pin-utils v0.1.0
    Checking smol v2.0.2
    Checking smol-hyper v0.1.1                                                                                                                                                                                     
    Checking http-body-util v0.1.2
 Documenting httparse v1.9.4                                                                                                                                                                                       
   Compiling bevy_state_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_state\macros)
   Compiling bevy_gizmos_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gizmos\macros)                                                                                                       
 Documenting futures-channel v0.3.30                                                                                                                                                                               
 Documenting tokio v1.40.0                                                                                                                                                                                         
    Checking bevy_reflect v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_reflect)                                                                                                                   
    Checking encase v0.10.0                                                                                                                                                                                        
    Checking bevy_mikktspace v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_mikktspace)                                                                                                             
    Checking hexasphere v15.0.0                                                                                                                                                                                    
 Documenting httpdate v1.0.3                                                                                                                                                                                       
 Documenting anyhow v1.0.89                                                                                                                                                                                        
    Checking bevy_ecs v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ecs)                                                                                                                           
    Checking bevy_math v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_math)                                                                                                                         
 Documenting encase_derive v0.10.0                                                                                                                                                                                 
 Documenting bevy_encase_derive v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_encase_derive)                                                                                                       
 Documenting rustix v0.38.37                                                                                                                                                                                       
    Checking bevy_color v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_color)                                                                                                                       
 Documenting bevy_gizmos_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gizmos\macros)                                                                                                       
    Checking bevy_app v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_app)                                                                                                                           
    Checking bevy_core v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_core)                                                                                                                         
    Checking bevy_input v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_input)                                                                                                                       
    Checking bevy_time v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_time)                                                                                                                         
    Checking bevy_hierarchy v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_hierarchy)                                                                                                               
    Checking bevy_a11y v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_a11y)                                                                                                                         
    Checking bevy_asset v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_asset)                                                                                                                       
    Checking bevy_log v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_log)                                                                                                                           
    Checking bevy_transform v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_transform)                                                                                                               
    Checking bevy_diagnostic v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_diagnostic)                                                                                                             
    Checking bevy_remote v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_remote)                                                                                                                     
    Checking bevy_state v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_state)                                                                                                                       
 Documenting bevy_state_macros v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_state\macros)                                                                                                         
    Checking bevy_window v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_window)                                                                                                                     
    Checking bevy_audio v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_audio)                                                                                                                       
    Checking bevy_winit v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_winit)                                                                                                                       
    Checking bevy_gilrs v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gilrs)                                                                                                                       
    Checking bevy_render v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_render)                                                                                                                     
 Documenting http v1.1.0                                                                                                                                                                                           
    Checking bevy_core_pipeline v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_core_pipeline)                                                                                                       
    Checking bevy_picking v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_picking)
    Checking bevy_scene v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_scene)
 Documenting bevy_reflect v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_reflect)                                                                                                                   
 Documenting bevy_mikktspace v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_mikktspace)
 Documenting hexasphere v15.0.0
warning: unresolved link to `crate::func::Function`                                                                                                                                                                
   --> crates\bevy_reflect\src\lib.rs:523:19
    |
523 | //! [`Function`]: crate::func::Function
    |                   ^^^^^^^^^^^^^^^^^^^^^ no item named `func` in module `bevy_reflect`
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

    Checking bevy_sprite v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_sprite)                                                                                                                     
    Checking bevy_pbr v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_pbr)
 Documenting futures-util v0.3.30                                                                                                                                                                                  
    Checking bevy_text v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_text)                                                                                                                         
    Checking bevy_ui v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ui)                                                                                                                             
    Checking bevy_gizmos v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gizmos)                                                                                                                     
    Checking bevy_animation v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_animation)                                                                                                               
    Checking bevy_gltf v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gltf)                                                                                                                         
    Checking bevy_internal v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_internal)                                                                                                                 
 Documenting encase v0.10.0                                                                                                                                                                                        
 Documenting async-io v2.3.4                                                                                                                                                                                       
 Documenting http-body v1.0.1                                                                                                                                                                                      
warning: `bevy_reflect` (lib doc) generated 1 warning
 Documenting bevy_ecs v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ecs)
 Documenting bevy_math v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_math)
 Documenting async-net v2.0.0
 Documenting async-process v2.3.0
 Documenting hyper v1.4.1
 Documenting http-body-util v0.1.2
 Documenting bevy_color v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_color)
 Documenting bevy_app v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_app)
 Documenting smol v2.0.2
 Documenting smol-hyper v0.1.1
 Documenting bevy_core v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_core)
 Documenting bevy_input v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_input)
 Documenting bevy_a11y v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_a11y)
 Documenting bevy_time v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_time)
 Documenting bevy_log v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_log)
 Documenting bevy_asset v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_asset)
 Documenting bevy_hierarchy v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_hierarchy)
 Documenting bevy_diagnostic v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_diagnostic)
 Documenting bevy_window v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_window)
 Documenting bevy_gilrs v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gilrs)
 Documenting bevy_transform v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_transform)
 Documenting bevy_state v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_state)
 Documenting bevy_remote v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_remote)
 Documenting bevy_winit v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_winit)
 Documenting bevy_audio v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_audio)
 Documenting bevy_render v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_render)
 Documenting bevy_core_pipeline v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_core_pipeline)
 Documenting bevy_picking v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_picking)
 Documenting bevy_scene v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_scene)
 Documenting bevy_sprite v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_sprite)
 Documenting bevy_pbr v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_pbr)
warning: unresolved link to `StandardMaterial::anisotropy_texture`
   --> crates\bevy_pbr\src\pbr_material.rs:505:11
    |
505 |     /// [`StandardMaterial::anisotropy_texture`].
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the struct `StandardMaterial` has no field or associated item named `anisotropy_texture`
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

 Documenting bevy_text v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_text)
warning: `bevy_pbr` (lib doc) generated 1 warning
 Documenting bevy_gizmos v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gizmos)
 Documenting bevy_ui v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_ui)
 Documenting bevy_animation v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_animation)
 Documenting bevy_gltf v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_gltf)
 Documenting bevy_internal v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy\crates\bevy_internal)
 Documenting bevy v0.15.0-dev (C:\Users\BenjaminBrienen\source\bevy)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 8m 59s
   Generated C:\Users\BenjaminBrienen\source\bevy\target\doc\bevy\index.html
PS C:\Users\BenjaminBrienen\source\bevy> set RUSTFLAGS='--cfg docsrs_dep' && set RUSTDOCFLAGS='--cfg=docsrs' && cargo +nightly doc
warning: unresolved link to `crate::func::Function`
   --> crates\bevy_reflect\src\lib.rs:523:19
    |
523 | //! [`Function`]: crate::func::Function
    |                   ^^^^^^^^^^^^^^^^^^^^^ no item named `func` in module `bevy_reflect`
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: `bevy_reflect` (lib doc) generated 1 warning
warning: unresolved link to `StandardMaterial::anisotropy_texture`
   --> crates\bevy_pbr\src\pbr_material.rs:505:11
    |
505 |     /// [`StandardMaterial::anisotropy_texture`].
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the struct `StandardMaterial` has no field or associated item named `anisotropy_texture`
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: `bevy_pbr` (lib doc) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.79s
   Generated C:\Users\BenjaminBrienen\source\bevy\target\doc\bevy\index.html
```

File: crates/bevy_reflect/src/func/reflect_fn_mut.rs (hunk 2 above)
Comment (BenjaminBrienen) [reply crates/bevy_reflect/src/func/reflect_fn_mut.rs:207]: It works for me with:

`set RUSTFLAGS='--cfg docsrs_dep' && set RUSTDOCFLAGS='--cfg=docsrs' && cargo +nightly doc`

**Review by bash — CHANGES_REQUESTED** [review]
The latest commit still fails for me with with various `error: #[doc(fake_variadic)] must be used on the first of a set of tuple or fn pointer trait impls with varying arity`s. 

I'm using the latest nightly (`2024-09-24`) and the following command to build: `RUSTFLAGS='--cfg docsrs_dep' RUSTDOCFLAGS='--cfg=docsrs' cargo +nightly doc`

Maybe the command that you used didn't correctly export the env vars?
> `set RUSTFLAGS='--cfg docsrs_dep' && set RUSTDOCFLAGS='--cfg=docsrs' && cargo +nightly doc`

If you're using fish you can either use the posix-style syntax that I used above or you have to use `set -x` to get the variable exported iirc.

**Comment by BenjaminBrienen** [comment]
> Maybe the command that you used didn't correctly export the env vars?

Ah, this command did the trick:
`$env:RUSTFLAGS='--cfg docsrs_dep'; $env:RUSTDOCFLAGS='--cfg=docsrs'; cargo +nightly doc`

Fixing the errors now!

**Comment by BenjaminBrienen** [comment]
The docs generate successfully for me now

**Comment by BenjaminBrienen** [comment]
image

File: crates/bevy_ecs/src/query/fetch.rs [hunk 3]
```diff
@@ -2008,11 +2008,10 @@ unsafe impl<T: Component> ReadOnlyQueryData for Has<T> {}
 pub struct AnyOf<T>(PhantomData<T>);
 
 macro_rules! impl_tuple_query_data {
-    ($(#[$meta:meta])* $(($name: ident, $state: ident)),*) => {
+    ($(($name: ident, $state: ident)),*) => {
 
         #[allow(non_snake_case)]
         #[allow(clippy::unused_unit)]
-        $(#[$meta])*
```
Comment (bash) [inline crates/bevy_ecs/src/query/fetch.rs:2015]: Why did you remove this? fake_variadic should work fine here

File: crates/bevy_reflect/src/func/reflect_fn.rs [hunk 4]
```diff
@@ -76,8 +76,9 @@ pub trait ReflectFn<'env, Marker>: ReflectFnMut<'env, Marker> {
 /// - `Fn(&mut Receiver, arg0, arg1, ..., argN) -> &mut R`
 /// - `Fn(&mut Receiver, arg0, arg1, ..., argN) -> &R`
 macro_rules! impl_reflect_fn {
-    ($(($Arg:ident, $arg:ident)),*) => {
```
Comment (bash) [inline crates/bevy_reflect/src/func/reflect_fn.rs:79]: I noticed that this didn't show up in the docs and found out that it's behind the `functions` feature. 
So I rebuilt with `--all-features` and got a `#[doc(fake_variadic)] must be used on the first of a set of tuple or fn pointer trait impls with varying arity` error :/

File: crates/bevy_reflect/src/func/reflect_fn_mut.rs [hunk 5]
```diff
@@ -82,8 +82,9 @@ pub trait ReflectFnMut<'env, Marker> {
 /// - `FnMut(&mut Receiver, arg0, arg1, ..., argN) -> &mut R`
 /// - `FnMut(&mut Receiver, arg0, arg1, ..., argN) -> &R`
 macro_rules! impl_reflect_fn_mut {
-    ($(($Arg:ident, $arg:ident)),*) => {
+    ($(#[$meta:meta])* $(($Arg:ident, $arg:ident)),*) => {
```
Comment (bash) [inline crates/bevy_reflect/src/func/reflect_fn_mut.rs:85]: Same as with `ReflectFn`

File: crates/bevy_app/src/plugin.rs [hunk 6]
```diff
@@ -162,7 +162,8 @@ mod sealed {
     }
 
     macro_rules! impl_plugins_tuples {
-        ($(($param: ident, $plugins: ident)),*) => {
+        ($(#[$meta:meta])* $(($param: ident, $plugins: ident)),*) => {
+            $(#[$meta])*
             impl<$($param, $plugins),*> Plugins<(PluginsTupleMarker, $($param,)*)> for ($($plugins,)*)
```
Comment (bash) [inline crates/bevy_app/src/plugin.rs:167]: This impl is for the internal `Plugins` trait, not the public one, so I think it doesn't show up in the docs.

File: crates/bevy_reflect/src/func/reflect_fn.rs (hunk 4 above)
Comment (BenjaminBrienen) [reply crates/bevy_reflect/src/func/reflect_fn.rs:79]: I'll fix that, thanks.

File: crates/bevy_app/src/plugin.rs (hunk 6 above)
Comment (BenjaminBrienen) [reply crates/bevy_app/src/plugin.rs:167]: `cargo doc` supports --document-private-items, to document private items, which is a useful tool for engine developers.

Comment (bash) [reply crates/bevy_app/src/plugin.rs:167]: Ah, right. Never mind my comment then :)

File: crates/bevy_ecs/src/query/fetch.rs [hunk 7]
```diff
@@ -2008,11 +2008,10 @@ unsafe impl<T: Component> ReadOnlyQueryData for Has<T> {}
 pub struct AnyOf<T>(PhantomData<T>);
 
 macro_rules! impl_tuple_query_data {
```
Comment (bash) [inline crates/bevy_ecs/src/query/fetch.rs:2010]: I think we misunderstood each other here :)

`impl QueryData for (...)` should be annotated with `fake_variadic`
`impl WorldQuery for AnyOf<(...)>` should not be annotated, so the macro no longer needs to acccept attributes via `#[$meta:meta])*`.

<sup>You can probably revert the entire file using `git restore --source=upstream/main crates/bevy_ecs/src/query/fetch.rs` assuming you have run `git fetch upstream` beforehand and the remote pointing to the bevy repo is called `upstream`.</sup>

**Review by bash — COMMENTED** [review]
lgtm otherwise ✨

**Review by 13ros27 — CHANGES_REQUESTED** [review]
Feel free to ignore (and mark resolved) the whitespace issues
EDIT: Sorry, didn't mean to request changes (I blame gh mobile lol)

**Comment by 13ros27** [comment]
I'm also unsure where `impl_system_function` came from as that appears to be new? I presume it was implemented somewhere before?

File: crates/bevy_app/src/lib.rs [hunk 8]
```diff
@@ -1,4 +1,6 @@
-#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+// `rustdoc_internals` is needed for `#[doc(fake_variadics)]`
+#![allow(internal_features)]
```
[the remark this answers is not in the document]
Comment (BenjaminBrienen) [reply crates/bevy_app/src/lib.rs:2]: That's what I have there, right?

**Comment by BenjaminBrienen** [comment]
Changes have been cleaned up. No idea how some of that happened lol.

File: crates/bevy_reflect/src/tuple.rs [hunk 9]
```diff
@@ -671,6 +671,7 @@ impl_reflect_tuple! {0: A, 1: B, 2: C, 3: D, 4: E, 5: F, 6: G, 7: H, 8: I, 9: J,
 
 macro_rules! impl_type_path_tuple {
     ($(#[$meta:meta])*) => {
+        $(#[$meta])*
```
Comment (BenjaminBrienen) [inline crates/bevy_reflect/src/tuple.rs:674]: I can remove this if wanted

**Comment by MrGVSV** [comment]
Can you verify that these are working for `IntoReturn`, `GetOwnership`, and `FromArg`? I'm not getting any errors, but my generated docs aren't rendering it properly.

File: crates/bevy_app/src/lib.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (13ros27) [reply crates/bevy_app/src/lib.rs:2]: I meant that maybe `internal_features` should also be under a `cfg_attr`, should have been clearer

**Review by 13ros27 — APPROVED** [review]
I haven't tried generating the docs myself (afk) but visually it all looks good to me, thanks for fixing the formatting

**Review by bushrat011899 — APPROVED** [review]
